component/src/test/java/com/celements/photo/container/ImageUrlTest.java

```diff
@@ -0,0 +1,225 @@
+    replayDefault();
+    assertEquals(DEF_DOCNAME, imgUrl.getName().get());
+    verifyDefault();
+  }
+
+  @Test
+  public void testGetFilename() throws Exception {
+    ImageUrl imgUrl = new ImageUrl.Builder().url(DEF_INT_URL).build();
+    replayDefault();
+    assertEquals(DEF_FILENAME, imgUrl.getFilename().get());
+    verifyDefault();
+  }
+
+  @Test
+  public void testGetQuery_empty() throws Exception {
+    ImageUrl imgUrl = new ImageUrl.Builder().url(DEF_INT_URL).build();
+    replayDefault();
+    assertEquals("", imgUrl.getQuery().get());
+    verifyDefault();
+  }
+
+  @Test
+  public void testGetQuery_withQuery() throws Exception {
+    String queryStr = "celwidth=1";
+    ImageUrl imgUrl = new ImageUrl.Builder().url(DEF_INT_URL + queryStr).build();
+    replayDefault();
+    assertEquals(queryStr, imgUrl.getQuery().get());
+    verifyDefault();
+  }
+
+  @Test
+  public void testGetWidth_none() throws Exception {
+    ImageUrl imgUrl = new ImageUrl.Builder().url(DEF_INT_URL).build();
+    replayDefault();
+    assertFalse(imgUrl.getWidth().isPresent());
+    verifyDefault();
+  }
+
+  @Test
+  public void testGetWidth_invalid() throws Exception {
+    ImageUrl imgUrl = new ImageUrl.Builder().url(DEF_INT_URL + "celwidth=hi").build();
+    replayDefault();
+    assertFalse(imgUrl.getWidth().isPresent());
+    verifyDefault();
+  }
+
+  @Test
+  public void testGetWidth_valid() throws Exception {
+    ImageUrl imgUrl = new ImageUrl.Builder().url(DEF_INT_URL + "celwidth=1001").build();
+    replayDefault();
+    assertEquals(1001, (int) imgUrl.getWidth().get());
+    verifyDefault();
+  }
+
+  @Test
+  public void testGetHeight_none() throws Exception {
+    ImageUrl imgUrl = new ImageUrl.Builder().url(DEF_INT_URL).build();
+    replayDefault();
+    assertFalse(imgUrl.getHeight().isPresent());
+    verifyDefault();
+  }
+
+  @Test
+  public void testGetHeight_invalid() throws Exception {
+    ImageUrl imgUrl = new ImageUrl.Builder().url(DEF_INT_URL + "celheight=9999999999999").build();
+    replayDefault();
+    assertFalse(imgUrl.getHeight().isPresent());
+    verifyDefault();
+  }
+
+  @Test
+  public void testGetHeight_valid() throws Exception {
+    ImageUrl imgUrl = new ImageUrl.Builder().url(DEF_INT_URL
+        + "celwidth=1001&celheight=999&other=x").build();
+    replayDefault();
+    assertEquals(999, (int) imgUrl.getHeight().get());
+    verifyDefault();
+  }
+
+  @Test
+  public void testToString() throws Exception {
+    ImageUrl imgUrl = new ImageUrl.Builder().url(DEF_INT_URL).build();
+    URL serverUrl = new URL(DEF_EXT_URL);
+    expect(urlFactory.createAttachmentURL(eq(DEF_FILENAME), eq(DEF_SPACE), eq(DEF_DOCNAME), eq(
+        DEF_ACTION), eq(""), eq(DEF_DB), eq(getContext()))).andReturn(serverUrl);
+    expect(urlFactory.getURL(eq(serverUrl), same(getContext()))).andReturn(DEF_INT_URL);
+    replayDefault();
+    assertEquals(DEF_INT_URL, imgUrl.getUrl());
+    verifyDefault();
+  }
+
+  // @Test
```

Comment:
If this new tests are not needed, please remove otherwise finalise implementation.